lib/HttpClient.ts

```diff
@@ -418,6 +418,7 @@ export class HttpClient implements ifm.IHttpClient {
         info.options.port = info.parsedUrl.port ? parseInt(info.parsedUrl.port) : defaultPort;
         info.options.path = (info.parsedUrl.pathname || '') + (info.parsedUrl.search || '');
         info.options.method = method;
+        info.options.timeout = this.requestOptions && this.requestOptions.socketTimeout;
```

Comment:
I think we want to also check (and set) `this._socketTimeout`. I think it should look something like

```
info.options.timeout = (this.requestOptions && this.requestOptions.socketTimeout) || this._socketTimeout;
this._socketTimeout = info.options.timeout;
```